mlir/include/mlir/Analysis/Presburger/Fraction.h

```diff
@@ -95,8 +99,52 @@ inline bool operator>=(const Fraction &x, const Fraction &y) {
   return compare(x, y) >= 0;
 }
 
+inline Fraction reduce(const Fraction &f) {
+  if (f == Fraction(0))
+    return f;
```

Comment:
Can you make this return Fraction(0)? The reason is that the denominator then gets simplified to 1.

Reply:
Actually make it Fraction(0, 1) so it's clear to the reader